packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +88,117 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
+            this.setState({ addCustom: true });
+        }
+
         this._localChange = true;
 
         const store = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         if (!this.props.noDirectUpdate) {
             this.props.target[this.props.propertyName] = this._remapValueOut(value as number);
         }
-        this.setState({ value: value });
 
-        if (this.props.onSelect) {
-            this.props.onSelect(value);
+        //selecting a regular option from font dropdown
+        if (value != 0 && this.props.fromFontDropdown) {
+            this.setState({ value: value });
+            if (this.props.onSelect) {
+                this.props.onSelect(value);
+            }
+            //selecting 'custom font' from font dropdown
+        } else if (this.props.fromFontDropdown) {
+            if (this.props.onSelect) {
+                this.props.onSelect(this.state.value);
+            }
+        }
+        //selecting from a dropdown that's not font dropdown
+        else {
+            this.setState({ value: value });
+            if (this.props.onSelect) {
+                this.props.onSelect(value);
+            }
         }
 
         const newValue = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         this.raiseOnPropertyChanged(newValue, store);
     }
 
+    updateCustomValue() {
+        this.setState({ addCustom: false });
+    }
+
     render() {
-        return (
-            <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
-                {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
-                <div className="label" title={this.props.label}>
-                    {this.props.label}
+        if (this.state.addCustom) {
+            return (
+                <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
+                    {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
+                    <div className="label" title={this.props.label}>
+                        {this.props.label}
+                    </div>
+                    <div className="options">
+                        <input
+                            type="text"
+                            placeholder="Enter a custom font here"
+                            id="customFont"
+                            onKeyDown={(event) => {
+                                event.key === "Enter" && this.props.addVal != undefined
+                                    ? (this.props.addVal(
+                                          { label: (document.getElementById("customFont") as HTMLInputElement).value, value: this.props.options.length + 1 },

```

Comment:
can this be fetched using a (react) reference? Otherwise this will require the application using it to have this specific id instead of passing a reference to the right html element.

Also - the id limits the data to be used (i.e. customFont is confusing. This can be used for other types of values, but this suggests it's only for custom fonts).

Reply:
I fixed this! I think Carol had a similar change so I don't use id anymore

Reply:
I just haven't committed yet